app/models/tag.rb

```diff
@@ -76,6 +76,20 @@ def self.aliased_name(word)
     tag.alias_for.presence || tag.name
   end
 
+  def self.find_alias_for(candidate)
```

Comment:
Why do we have multiple levels of aliases? This is not going to be very performant and arbitrarily going 4 deep feels a little weird. I think we should be eliminating the in-between aliases if the goal is always to arrive at the lowest common denominator.

Reply:
That actually is an arbitrary number I thought of as I extracted the code from [TagParser ](https://github.com/thepracticaldev/dev.to/blob/master/app/lib/acts_as_taggable_on/tag_parser.rb#L20) in case we ever accidentally looped an alias onto itself. 

The original idea was we would swap in a more preferred tag and it would be more straight forward to apply new alias onto the existing preferred tag instead of changing the alias of the tail. We were definitely wrong there and it should've been the other way around but this isn't a cause for concern as there is not a lot of Tags that currently have an alias. Would it be better if I lower that number down to 2?

Reply:
If we have a low number of "alias chains" then I think we should jump in a console and remove the alias chains in favor of a single alias and then we don't need this looping logic. What do you think?

Reply:
Let's do that